### ISSUE TITLE
Switch from awslogs driver to FireLens and FluentBit

### ISSUE THREAD

**Issue by BowenRaymone** [aws/aws-for-fluent-bit#136]
Hi,

We have recently switched to use FireLens as log driver, instead of awslogs driver. After the switch, I can see the benefits of using FireLens with more customization and even latency reduce (surprisingly good side effect). However, I also noticed some disadvantages, and would like to see if we can avoid them. 

First of all, FireLens driver will silently drop logs when the log line is huge for input type 'tcp'. For example, I have a log line with ~200,000 characters (200K), and I found that FluentBit simply silently not publish to CWL. I had to research several parameters, and finally adjusted `Chunk_Size` and `Buffer_Size` parameters to be able to publish those logs. In contrast, awslogs driver will send them anyway. I realized that it might be FluentBit's issue, but I wonder if we can provide more error or warning message for log drop. And once we saw those messages, we can tune parameters and react accordingly. 

Also, to gain more confidence about preserving all logs, is there a way that we can monitor log drop rate? 

Thanks for your help!

Best,
- Bowen

**Comment by hossain-rayhan** [comment]
Hi @BowenRaymone, If I remember correctly, our plugins should print some kind of warning message and send the first allowed number of characters in case of huge log line. However, it would be easier for us to track if you provide the whole configuration file you are using.

**Comment by BowenRaymone** [comment]
Sure, I didn't see any warning message and the first few characters were not published using my config. After I increase `Chunk_Size` and `Buffer_Size` to 256, I can start to see those in logs. 

Here is the config I use:
```
[SERVICE]
    Flush                       1
    Grace                       30
    Log_Level                   debug
    Parsers_File                /fluent-bit-config/parsers.conf

[INPUT]
    Name            tcp
    Tag             ApplicationLogs
    Listen          0.0.0.0
    Port            5170
    Chunk_Size      256
    Buffer_Size     256
    Format          None
    Mem_Buf_Limit   800MB

[OUTPUT]
    Name               cloudwatch_logs
    Match              ApplicationLogs
    region             ${LOG_REGION}
    log_group_name     ${SERVICE_NAME}/application_log
    log_stream_name    application_log_${LOG_STREAM_SUFFIX}
    log_key            log
    auto_create_group  true
```

Please let me know how I can improve on this. Thanks!

**Comment by hossain-rayhan** [comment]
Oh I see. We have this feature implemented in the `cloudwatch` output plugin (CloudWatch Go Plugin) [link](https://github.com/aws/amazon-cloudwatch-logs-for-fluent-bit/blob/mainline/cloudwatch/cloudwatch.go#L642). Thanks for the configuration. We will take a look to support this in our `cloudwatch_logs` output plugin (CloudWatch C Plugin).

**Comment by PettitWesley** [comment]
Actually I think Rayhan's suggestion isn't what is needed...

I believe the CloudWatch C plugin already has a warning message if it has to drop a log message because it's too long. Will check.

It sounds like changing the `Chunk_size` and `Buffer_size` in the TCP input is what fixed this problem... so your ask is actually to have some sort of warning message in that plugin for dropped events.

**Comment by BowenRaymone** [comment]
Yes, I want to check and see if there's any warning message for dropped events so that we will know in the future and react properly. Otherwise, we won't know if all logs are published or not. 

Another option is whether there is any metric that we can alarm on so that we can monitor and react. 

Thanks!

**Comment by ShelbyZ** [comment]
Detecting dropped logs should not be as much of an issue at this current time. There are both guides on log-loss, https://github.com/aws/aws-for-fluent-bit/blob/mainline/troubleshooting/debugging.md#log-loss and more logging from plugins that makes checking a bit easier.

I am going to leave this open for now, but without correspondence expect to close by 6/30/2025
